benchmark_v2/reformat_benchmark_results.py

```diff
@@ -0,0 +1,415 @@
+    parser.add_argument(
+        "--graph-infra",
+        default=os.path.join(benchmark_dir, "graph_infra"),
+        help="Directory containing graph infrastructure",
+    )
+    parser.add_argument("--platform-name", default="", help="Test platform name")
+    parser.add_argument("--description", default="", help="Description of this configuration")
+    parser.add_argument("--annotations", default="", help="File containing annotations")
+    parser.add_argument("--configs", help="comma seperate list of configurations")
+    parser.add_argument("--start-date", help="graph start date")
+    parser.add_argument("--benchmark-data", help="the benchnmark output data in json format.")
+
+    return parser
+
+
+def get_header_dict(directory_path):
+    """Get a dictionary there the keys are the benchmark name and the values are list of header fields for the .dat file."""
+    headers = read_files_in_directory(directory_path)
+    for benchmark_name in BENCHMARKS:
+        if benchmark_name not in headers.keys():
+            headers[benchmark_name] = ["Average time =", "Average rate ="]
+
+    for key in headers.keys():
+        headers[key].insert(0, "# Date")
+
+    return headers
+
+
+# This algorithm for reading files into a dictionary was generated with assistance from Perplexity AI (2024).
+def read_files_in_directory(directory_path):
+    """Get a dictionary there the keys are the benchmark name and the values are list of header fields for the .dat file.
+    Only retrieves header fields in the .perfkeys files in the graph_infra directory."""
+    # Dictionary to store file names and their contents
+    file_contents = {}
+
+    # Iterate through all files in the directory
+    for filename in os.listdir(directory_path):
+        file_path = os.path.join(directory_path, filename)
+
+        # Check if it's a file (not a directory)
+        if (".perfkeys" in file_path) and os.path.isfile(file_path):
+            try:
+                # Open and read the file
+                with open(file_path, "r", encoding="utf-8") as file:
+                    # Read all lines and store them in a list
+                    lines = file.readlines()
+
+                    # Strip newline characters from each line
+                    lines = [line.strip() for line in lines]
+
+                    # Add the file name and its contents to the dictionary
+                    key = re.search(r"([\w_]+).perfkeys", filename)[1]
+                    file_contents[key] = lines
+            except Exception as e:
+                print(f"Error reading file {filename}: {str(e)}")
+
+    return file_contents
+
+
+def get_nested_value(data: dict, keys: list):
+    """Look up a value in a dictionary using a list of keys."""
+    for key in keys:
+        if isinstance(data, dict):
+            data = data.get(key, {})
+        else:
+            return None
+    return data if data != {} else None
+
```

Comment:
This feels a bit tricky to me, so let me see if I understand it.
Once you find a key, you change data to the value data.get(key,{}), and it's no longer a dictionary.
That effectively ends your for loop.
Is that accurate?
If so, (1) it's not the way I would do it, but (2) approval isn't a matter of "did you do it the way I would?"

But since I find this a bit confusing, I'd appreciate maybe some comments making it just a bit clearer.